src/counts.jl

```diff
@@ -110,26 +121,25 @@ proportions(x::IntegerArray, wv::AbstractWeights) = proportions(x, span(x), wv)
 #### functions for counting a single list of integers (2D)
 
 function addcounts!(r::AbstractArray, x::IntegerArray, y::IntegerArray, levels::NTuple{2,IntUnitRange})
-    # add counts of integers from x to r
-
-    n = length(x)
-    length(y) == n || throw(DimensionMismatch())
+    # add counts of pairs from zip(x,y) to r
 
     xlevels, ylevels = levels
 
-    kx = length(xlevels)
-    ky = length(ylevels)
-    size(r) == (kx, ky) || throw(DimensionMismatch())
+    axes(x) == axes(y) ||
+        throw(DimensionMismatch("x and y must have the same axes, but got $(axes(x)) and $(axes(y))"))
 
-    mx0 = xlevels[1]
-    mx1 = xlevels[end]
-    my0 = ylevels[1]
-    my1 = ylevels[end]
+    axes(r) == (axes(xlevels, 1), axes(ylevels, 1)) ||
+        throw(DimensionMismatch("axes(r) must correspond to the axes of levels, but got $(axes(r)) ≠ $(axes(xlevels, 1)), $(axes(ylevels, 1))"))
+
+    mx0 = first(xlevels)
+    mx1 = last(xlevels)
+    my0 = first(ylevels)
+    my1 = last(ylevels)
 
     bx = mx0 - 1
     by = my0 - 1
 
-    for i = 1:n
+    for i in eachindex(x, y)
```

Comment:
This is supposed to work any time `x` and `y` have the same length. Failing on OffsetVectors with different offsets is acceptable. Failing on row matrix + column matrix is not.
```suggestion
    for i in eachindex(vec(x), vec(y))
```